auction-server/src/auction.rs

```diff
@@ -1066,3 +1016,367 @@ async fn simulate_bid_svm(chain_store: &ChainStoreSvm, bid: &BidSvm) -> Result<(
         None => Ok(()),
     }
 }
+
+/// The trait for the chain store to be implemented for each chain type
+/// These functions are chain specific and should be implemented for each chain in order to handle auctions
+pub trait ChainStore {
+    /// The block type for the chain
+    type Block: DebugTrait;
+    /// The block stream type when subscribing to new blocks on the ws client for the chain
+    type BlockStream<'a>: Stream<Item = Self::Block> + Unpin + Send + 'a;
+    /// The ws client type for the chain
+    type WsClient;
+    /// The simulated bid type for the chain
+    type SimulatedBid: SimulatedBidTrait;
+    /// The conclusion result type when try to conclude the auction for the chain
+    type ConclusionResult;
+
+    /// The chain type for the chain
+    const CHAIN_TYPE: models::ChainType;
+    /// The minimum lifetime for an auction. If any bid for auction is older than this, the auction is ready to be submitted.
+    const AUCTION_MINIMUM_LIFETIME: Duration;
+
+    /// Get the ws client for the chain
+    fn get_ws_client(self) -> impl Future<Output = Result<Self::WsClient>> + Send;
+    /// Get the block stream for the ws client to subscribe to new blocks
+    fn get_block_stream<'a>(
+        client: &'a Self::WsClient,
+    ) -> impl Future<Output = Result<Self::BlockStream<'a>>>;
+    /// Filter the bids to only include the chain specific bids
+    fn filter_bids(bids: Vec<SimulatedBid>) -> Vec<Self::SimulatedBid>;
```

Comment:
I think we only need this function to convert types and we are certain that all the input bids are for that specific chain. So maybe rename it to convert_bids and panic if the input is not valid?